python/lsst/analysis/drp/calcFunctors.py

```diff
@@ -19,3 +21,20 @@ def setDefaults(self):
         super().setDefaults()
         self.colA.column = "iKronFlux"
         self.colB.column = "iPsFlux"
+
+
+class MagDiff(MultiColumnAction):
+    """Calculate the difference between two magntiudes"""
+
+    col1 = Field(doc="Column to subtract from", dtype=str)
+    col2 = Field(doc="Column to subtract", dtype=str)
+
+    @property
+    def columns(self):
+        return (self.col1, self.col2)
+
+    def __call__(self, df):
```

Comment:
This has assumptions of units built into it. Can you please use astropy quantities and be more explicit about what units you expect the columns to have? Logging something like "subtracting col2 from col1 assuming flux units of X" might be worthwhile, for example.

Reply:
I have added additional docs and used a config option for astropy units for the fluxes. Astropy doesn't have a jansky -> magnitude option nor does it have a millimag option so that doesn't really work for the outputs. The number of units that can be given is also relatively small as this action doesn't take a photocalib or any zeropoint information because it is expected to convert janskies to magnitudes. I will make this explicit in the docs.

Reply:
This is what @parejkoj recommended in a Community post a little while back, that I always refer to, if it helps
```
>>> import astropy.units as u
>>> (575*u.nJy).to(u.ABmag)
<Magnitude 24.50083039 mag(AB)>
>>> (0*u.ABmag).to_value(u.nJy)
3630780547701.003
```

Reply:
Also `3631.0` is an approximation at a level that matters for our precision level. Just use astropy, as @mrawls says. (do the input/output data have units attached? If not, they really really should)

Reply:
Somehow I missed the magAB option in the list of units. Thanks guys!